components/io_expander/esp_io_expander/include/esp_io_expander.h

```diff
@@ -19,15 +19,60 @@ extern "C" {
  * @brief IO Expander Device Type
  *
  */
-typedef struct esp_io_expander_t esp_io_expander_t;
+typedef struct esp_io_expander_s esp_io_expander_t;
 typedef esp_io_expander_t *esp_io_expander_handle_t;
 
+/**
+ * @brief IO Expander Pin Num
+ *
+ */
+typedef enum {
+    IO_EXPANDER_PIN_NUM_0  = 0,
+    IO_EXPANDER_PIN_NUM_1  = 1,
+    IO_EXPANDER_PIN_NUM_2  = 2,
+    IO_EXPANDER_PIN_NUM_3  = 3,
+    IO_EXPANDER_PIN_NUM_4  = 4,
+    IO_EXPANDER_PIN_NUM_5  = 5,
+    IO_EXPANDER_PIN_NUM_6  = 6,
+    IO_EXPANDER_PIN_NUM_7  = 7,
+    IO_EXPANDER_PIN_NUM_8  = 8,
+    IO_EXPANDER_PIN_NUM_9  = 9,
+    IO_EXPANDER_PIN_NUM_10 = 10,
+    IO_EXPANDER_PIN_NUM_11 = 11,
+    IO_EXPANDER_PIN_NUM_12 = 12,
+    IO_EXPANDER_PIN_NUM_13 = 13,
+    IO_EXPANDER_PIN_NUM_14 = 14,
+    IO_EXPANDER_PIN_NUM_15 = 15,
+    IO_EXPANDER_PIN_NUM_16 = 16,
+    IO_EXPANDER_PIN_NUM_17 = 17,
+    IO_EXPANDER_PIN_NUM_18 = 18,
+    IO_EXPANDER_PIN_NUM_19 = 19,
+    IO_EXPANDER_PIN_NUM_20 = 20,
+    IO_EXPANDER_PIN_NUM_21 = 21,
+    IO_EXPANDER_PIN_NUM_26 = 26,
+    IO_EXPANDER_PIN_NUM_27 = 27,
+    IO_EXPANDER_PIN_NUM_28 = 28,
+    IO_EXPANDER_PIN_NUM_29 = 29,
+    IO_EXPANDER_PIN_NUM_30 = 30,
+    IO_EXPANDER_PIN_NUM_31 = 31,
+    IO_EXPANDER_PIN_NUM_MAX,
+} esp_io_expander_pin_num_t;
+
+/**
+ * @brief IO Expander Pin direction
+ *
+ */
+typedef enum {
+    IO_EXPANDER_INPUT,          /*!< Input direction */
+    IO_EXPANDER_OUTPUT,         /*!< Output dircetion */
+} esp_io_expander_dir_t;
+
 /**
  * @brief IO Expander Configuration Type
  *
  */
 typedef struct {
-    uint8_t io_amount;                      /*!< Amout of device's IO, must be smaller than 32 */
+    uint8_t io_amount;                      /*!< Amout of device's IO, must be less than `IO_EXPANDER_PIN_NUM_MAX` */
```

Comment:
```suggestion
    uint8_t io_count;                      /*!< Count of device's IO, must be less or equal than `IO_EXPANDER_PIN_NUM_MAX` */
```

Please use 'count' with countable nouns.

Should it be less or equal? `io_count = 32;` is valid, isn't it?